### PR TITLE
check if user is root before allowing build from definition

### PR DESCRIPTION
**Pull request by ikaneshiro**
Title

### CODE REVIEW

File: src/pkg/build/build.go [hunk 1]
```diff
@@ -292,6 +292,12 @@ func makeDef(spec string) (types.Definition, error) {
 		}
 
 	} else if ok, err := types.IsValidDefinition(spec); ok && err == nil {
+
+		//must be root to build froma definition
```
Comment (ArangoGutierrez) [inline src/pkg/build/build.go]: typo `froma`

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]

File: src/pkg/build/build.go (hunk 1 above)
Comment (ikaneshiro) [reply src/pkg/build/build.go]: Looks like i cant type today

Comment (ArangoGutierrez) [reply src/pkg/build/build.go]: :smile:

File: src/pkg/build/build.go [hunk 2]
```diff
@@ -292,6 +292,12 @@ func makeDef(spec string) (types.Definition, error) {
 		}
 
 	} else if ok, err := types.IsValidDefinition(spec); ok && err == nil {
+
+		//must be root to build from a definition
```
Comment (ArangoGutierrez) [inline src/pkg/build/build.go]: niptick 
```
// must be root to build from a definition
```

as you can see in the next comment ;)

**Review by ArangoGutierrez — COMMENTED** [review]
small niptick

**Review by ArangoGutierrez — APPROVED** [review]
```bash
[eduardo@rhel7 singularity]$ singularity build ian.sif examples/docker/Singularity
FATAL:   You must be the root user to build from a Singularity recipe file
[eduardo@rhel7 singularity]$ sudo singularity build ian.sif examples/docker/Singularity
WARNING: Authentication token file not found : Only pulls of public images will succeed
INFO:    Running %pre script
INFO:    Finished running %pre script. exit status 0
Getting image source signatures
Skipping fetch of repeat blob sha256:8c5a7da1afbc602695fcb2cd6445743cec5ff32053ea589ea9bd8773b7068185
Copying config sha256:665b3432b40dd40571ac4891b2dcbaaa7d39c69eaf7d6bff3a9e25acd63f4610
 575 B / 575 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
INFO:    Running %setup script
INFO:    Finished running %setup script. exit status 0
INFO:    Running %post script
Hello from inside the container
Install additional software here
INFO:    Finished running %post script. exit status 0
Parallel mksquashfs: Using 8 processors
Creating 4.0 filesystem on /tmp/sbuild-oci-568601552/squashfs-322860548.img, block size 131072.
[==========================================================================================================================================================================================|] 31/31 100%

Exportable Squashfs 4.0 filesystem, gzip compressed, data block size 131072
        compressed data, compressed metadata, compressed fragments, compressed xattrs
        duplicates are removed
Filesystem size 694.56 Kbytes (0.68 Mbytes)
        60.29% of uncompressed filesystem size (1151.99 Kbytes)
Inode table size 669 bytes (0.65 Kbytes)
        24.12% of uncompressed inode table size (2774 bytes)
Directory table size 2464 bytes (2.41 Kbytes)
        38.44% of uncompressed directory table size (6410 bytes)
Xattr table size 53 bytes (0.05 Kbytes)
        100.00% of uncompressed xattr table size (53 bytes)
Number of duplicate files found 4
Number of inodes 56
Number of files 25
Number of fragments 1
Number of symbolic links  6
Number of device nodes 0
Number of fifo nodes 0
Number of socket nodes 0
Number of directories 25
Number of ids (unique uids + gids) 1
Number of uids 1
        root (0)
Number of gids 1
        root (0)
```

**Pull request merged**